be/src/env/env_hdfs.cpp

```diff
@@ -38,44 +38,52 @@ void HdfsRandomAccessFile::close() noexcept {
     }
 }
 
-static Status read_at_internal(hdfsFS fs, hdfsFile file, const std::string& file_name, int64_t offset, Slice* res) {
+static Status read_at_internal(hdfsFS fs, hdfsFile file, const std::string& file_name, int64_t offset, Slice* res,
+                               bool usePread) {
     auto cur_offset = hdfsTell(fs, file);
     if (cur_offset == -1) {
         return Status::IOError(
                 strings::Substitute("fail to get offset, file=$0, error=$1", file_name, get_hdfs_err_msg()));
     }
     if (cur_offset != offset) {
         if (hdfsSeek(fs, file, offset)) {
-            return Status::IOError(strings::Substitute("fail to seek offset, file=$0, offset=$1, error=$1", file_name,
+            return Status::IOError(strings::Substitute("fail to seek offset, file=$0, offset=$1, error=$2", file_name,
                                                        offset, get_hdfs_err_msg()));
         }
     }
-    size_t bytes_read = 0;
-    while (bytes_read < res->size) {
-        size_t to_read = res->size - bytes_read;
-        auto hdfs_res = hdfsRead(fs, file, res->data + bytes_read, to_read);
-        if (hdfs_res < 0) {
-            return Status::IOError(
-                    strings::Substitute("fail to read file, file=$0, error=$1", file_name, get_hdfs_err_msg()));
-        } else if (hdfs_res == 0) {
-            break;
+    if (usePread) {
+        if (hdfsPreadFully(fs, file, offset, res->data, res->size) == -1) {
```

Comment:
What's the difference between hdfsPreadFully and hdfsRead?
Could all use the hdfsPreadFully instead of hdfsRead?

Reply:
Better to comment in this files to let people know why do it differently.

Reply:
remove hdfsRead and keep hdfsPreadFully only after discussion offline. PTAL